rust-mode.el

```diff
@@ -260,18 +260,88 @@ to the function arguments.  When nil, `->' will be indented one level."
         ;; Rewind until the point no longer moves
         (setq continue (/= starting (point)))))))
 
-(defun rust-in-macro ()
+(defvar-local rust-macro-scopes nil
+  "Cache for the scopes calculated by `rust-macro-scope'.
+
+This variable can be `let' bound directly or indirectly around
+`rust-macro-scope' as an optimization but shoult not be otherwise
+set.")
+
+(defun rust-macro-scope (&optional start end)
```

Comment:
if all call sites use the arguments, why not make them mandatory?

Reply:
Yes. Actually, they were effectively mandatory because I used `end` without a nil check.